server/container/guice/memory-guice/src/test/java/org/apache/james/GuiceJamesServerStartUpCheckTest.java

```diff
@@ -0,0 +1,176 @@
+                .combineWith(MemoryJamesServerMain.IN_MEMORY_SERVER_AGGREGATE_MODULE)
+                .overrideWith(new TestJMAPServerModule(LIMIT_TO_10_MESSAGES))
+                .overrideWith(binder -> binder.bind(TextExtractor.class).to(PDFTextExtractor.class))
+                .overrideWith(binder -> Multibinder
+                    .newSetBinder(binder, StartUpCheck.class)
+                    .addBinding().to(NoopStartUpCheck.class)))
+            .disableAutoStart()
+            .build();
+    }
+
+    @Nested
+    class WithStartUpCheckRequireGuiceComponents implements StartUpCheckSuccessContract {
+
+        @RegisterExtension
+        JamesServerExtension jamesServerExtension = new JamesServerBuilder()
+            .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
+                .combineWith(MemoryJamesServerMain.IN_MEMORY_SERVER_AGGREGATE_MODULE)
+                .overrideWith(new TestJMAPServerModule(LIMIT_TO_10_MESSAGES))
+                .overrideWith(binder -> binder.bind(TextExtractor.class).to(PDFTextExtractor.class))
+                .overrideWith(binder -> Multibinder
+                    .newSetBinder(binder, StartUpCheck.class)
+                    .addBinding().to(TestBlobExportMechanismStartUpCheck.class)))
+            .disableAutoStart()
+            .build();
+    }
+
+    @Nested
+    class WithNoStartUpCheck implements StartUpCheckSuccessContract {
+
+        @RegisterExtension
+        JamesServerExtension jamesServerExtension = new JamesServerBuilder()
+            .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
+                .combineWith(MemoryJamesServerMain.IN_MEMORY_SERVER_AGGREGATE_MODULE)
+                .overrideWith(new TestJMAPServerModule(LIMIT_TO_10_MESSAGES))
+                .overrideWith(binder -> binder.bind(TextExtractor.class).to(PDFTextExtractor.class)))
+            .disableAutoStart()
+            .build();
+    }
+
+    @Nested
+    class StartUpCheckFails {
+
+        @RegisterExtension
+        JamesServerExtension jamesServerExtension = new JamesServerBuilder()
+            .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
+                .combineWith(MemoryJamesServerMain.IN_MEMORY_SERVER_AGGREGATE_MODULE)
+                .overrideWith(new TestJMAPServerModule(LIMIT_TO_10_MESSAGES))
+                .overrideWith(binder -> binder.bind(TextExtractor.class).to(PDFTextExtractor.class))
+                .overrideWith(binder -> {
+                    Multibinder<StartUpCheck> setBinder = Multibinder
+                        .newSetBinder(binder, StartUpCheck.class);
+
+                    setBinder.addBinding().to(NoopStartUpCheck.class);
+                    setBinder.addBinding().to(FailingStartUpCheck.class);
+                }))
+            .disableAutoStart()
+            .build();
+
+        @Test
+        void serverShouldThrowWhenAStartUpCheckFails(GuiceJamesServer server) throws Exception {
```

Comment:
This test is redundant with `startUpCheckFailsShouldThrowAnExceptionCarryingOnlyBadChecks` just below and can be removed IMO